browser/src/control/Control.FormulaBarJSDialog.js

```diff
@@ -32,61 +32,79 @@ L.Control.FormulaBarJSDialog = L.Control.extend({
 	},
 
 	createFormulabar: function(text) {
-		var data = [
-			{
-				type: 'toolbox',
-				children: [
-					{
-						id: 'functiondialog',
-						type: 'toolitem',
-						text: _('Function Wizard'),
-						command: '.uno:FunctionDialog'
-					},
-					(!window.mode.isMobile()) ? (
+		if (!window.mode.isMobile()) {
+			var data = [
+				{
+					type: 'toolbox',
+					children: [
+						{
+							id: 'functiondialog',
+							type: 'toolitem',
+							text: _('Function Wizard'),
+							command: '.uno:FunctionDialog'
+						},
 						{
 							id: 'autosummenu:AutoSumMenu',
 							type: 'menubutton',
 							command: '.uno:AutoSumMenu'
-						}
-					) : {},
-					{
-						id: 'startformula',
-						type: 'customtoolitem',
-						text: _('Formula'),
-					},
-					// on mobile we show other buttons on the top bar
-					(!window.mode.isMobile()) ? (
+						},
+						{
+							id: 'startformula',
+							type: 'customtoolitem',
+							text: _('Formula'),
+						},
 						{
+							// on mobile we show other buttons on the top bar
 							id: 'acceptformula',
 							type: 'customtoolitem',
 							text: _('Accept'),
 							visible: false
-						}
-					) : {},
-					(!window.mode.isMobile()) ? (
+						},
 						{
 							id: 'cancelformula',
 							type: 'customtoolitem',
 							text: _UNO('.uno:Cancel', 'spreadsheet'),
 							visible: false
-						}
-					) : {},
-					{
-						id: 'sc_input_window',
-						type: 'multilineedit',
-						text: text ? text : '',
-						rawKeyEvents: window.mode.isDesktop() ? true : undefined,
-						useTextInput: window.mode.isDesktop() ? undefined : true
-					},
-					{
-						id: 'expand',
-						type: 'pushbutton',
-						text: '',
-						symbol: 'SPIN_DOWN',
-					}
-				]
-			}
-		];
+						},
+						{
+							id: 'sc_input_window',
+							type: 'multilineedit',
+							text: text ? text : '',
+							rawKeyEvents: window.mode.isDesktop() ? true : undefined,
```

Comment:
remove condition here it is always non mobile

Reply:
what about tablet? shouldn't we leave those conditions for seTextInput and rawKeyEvents in this case?

Reply:
ah, yes you are right. it should be as is here